### PR TITLE
moved custom_spec['ipsettings'] section inside check for key

### LINKED ISSUE #41: vmware template deploy failing when using nicSettingMap
Had to chop up some code to get this to work, so seems like a bug to me.  Here's my cloudinit for deploying a Vmware template with a static ip:

`LinuxPrep:
domain: <%= @host.domain %>
hostname: <%= @host.shortname%>
hwClockUTC: true
fqdn: <%= @host %>
ssh_pwauth: <%= @host.params['ssh_pwauth'] || true %>
timeZone: <%= @host.params['time-zone'] || 'UTC' %>

globalIPSettings: 
  dnsSuffixList: [<%= @host.domain %>] 

nicSettingMap:
<% @host.interfaces.each do |interface| %>
- adapter:
    dnsDomain: <%= interface.domain %>
    dnsServerList: [<%= interface.subnet.dns_primary %>, <%= interface.subnet.dns_secondary %>]
    gateway: [<%= interface.subnet.gateway %>]
    ip: <%= interface.ip %>
    subnetMask: <%= interface.subnet.mask %>
<% end %>`

A ipsettings section is not required for this to work properly.  Looking through Fog's cloudinit_to_customspec, it understands this, and only sets ipsettings hash IF it exists.  In this case, it's not set nor required, since we're doing it on a per adapter basis with nicSettingMap

`custom_spec['nicSettingMap']    = user_data['nicSettingMap'] if user_data.key?('nicSettingMap')
custom_spec['ipsettings']                  =  { 'ip' => user_data['ip'] } if user_data.key?('ip')`

However in vsphere/requests/compute/vm_clone.rb, around line 265, it's doing a `custom_spec['ipsettings'].key?`, which results in a nil call because ipsettings isn't actually set.

### PR DESCRIPTION
**Pull request by jberry**
fixes #41

### CODE REVIEW

File: lib/fog/vsphere/requests/compute/vm_clone.rb [hunk 1]
```diff
@@ -268,21 +268,21 @@ def vm_clone(options = {})
                 custom_spec['globalIPSettings']=Hash.new unless custom_spec.key?('globalIPSettings')
                 custom_spec['globalIPSettings']['dnsServerList'] = custom_spec['ipsettings']['dnsServerList'] if custom_spec['ipsettings'].key?('dnsServerList')
                 custom_spec['globalIPSettings']['dnsSuffixList'] = custom_spec['dnsSuffixList'] || [custom_spec['domain']] if ( custom_spec['dnsSuffixList'] || custom_spec['domain'])
+                if (custom_spec['ipsettings'].key?('ip') or custom_spec['ipsettings'].key?('gateway') or custom_spec['ipsettings'].key?('subnetMask') or custom_spec['ipsettings'].key?('domain') or custom_spec['ipsettings'].key?('dnsServerList'))
+                    if custom_spec['ipsettings'].key?('ip') 
+                      raise ArgumentError, "subnetMask is required for static ip" unless custom_spec["ipsettings"].key?("subnetMask")
```
Comment (plribeiro3000) [inline lib/fog/vsphere/requests/compute/vm_clone.rb]: Maybe you can combine this inline unless with the parent if conditional to improve readability?

File: lib/fog/vsphere/requests/compute/vm_clone.rb [hunk 2]
```diff
@@ -268,21 +268,21 @@ def vm_clone(options = {})
...
+                    custom_spec['nicSettingMap']=Array.new unless custom_spec.key?('nicSettingMap')
+                    custom_spec['nicSettingMap'][0]=Hash.new unless custom_spec['nicSettingMap'].length > 0
+                    custom_spec['nicSettingMap'][0]['adapter']=Hash.new unless custom_spec['nicSettingMap'][0].key?('adapter')
+                    custom_spec['nicSettingMap'][0]['adapter']['ip'] = custom_spec['ipsettings']['ip'] if custom_spec['ipsettings'].key?('ip')
+                    custom_spec['nicSettingMap'][0]['adapter']['gateway'] = custom_spec['ipsettings']['gateway'] if custom_spec['ipsettings'].key?('gateway')
+                    custom_spec['nicSettingMap'][0]['adapter']['subnetMask'] = custom_spec['ipsettings']['subnetMask'] if custom_spec['ipsettings'].key?('subnetMask')
+                    custom_spec['nicSettingMap'][0]['adapter']['dnsDomain'] = custom_spec['ipsettings']['domain'] if custom_spec['ipsettings'].key?('domain')
+                    custom_spec['nicSettingMap'][0]['adapter']['dnsServerList'] = custom_spec['ipsettings']['dnsServerList'] if custom_spec['ipsettings'].key?('dnsServerList')
+                 end       
               end
```
Comment (jrgarcia) [inline lib/fog/vsphere/requests/compute/vm_clone.rb]: I'd also remove this empty line. Not a big deal, but it helps with following the nesting.

**Comment by jberry** [comment]
Moved inline unless into parent if statement for easier readability per plribeiro3000
Removed empty line per jrgarcia
Fixed end alignment to please codeclimate

**Comment by plribeiro3000** [comment]
Code wise LGTM. @jrgarcia have more insights on the platform to move this forward!

**Comment by jrgarcia** [comment]
@plribeiro3000 Travis is failing the 1.9.x builds. It's because of dependencies though. I'm installing 1.9.3 on my machine right now. If just pinning those dependencies fixes things, I'll merge this in and then update the gemspec with those dependencies pinned and release a new version of fog-vsphere. @plribeiro3000 What do you think about making a 1.0.0 release soon?

**Comment by jberry** [comment]
For what it's worth, I didn't add or remove any existing code (with the exception of that change of the inline unless into the if statement).  

All I did was move an existing if block inside another existing if block to prevent the nil case

**Comment by jrgarcia** [comment]
Well, everything seems to run just fine locally with 1.9.3 and JRuby in 1.9 mode.

**Comment by plribeiro3000** [comment]
@jrgarcia Go ahead with the `1.0.0` release. 👍

**Pull request merged**